Vostok.Hosting/Components/Log/LogLevelMetrics.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using Vostok.Commons.Time;
+using Vostok.Hosting.Abstractions;
+using Vostok.Logging.Abstractions;
+using Vostok.Metrics;
+using Vostok.Metrics.Models;
+using Vostok.Metrics.Primitives.Gauge;
+
+namespace Vostok.Hosting.Components.Log
+{
+    internal class LogLevelMetrics
+    {
+        private static readonly TimeSpan ScrapePeriod = 1.Minutes();
+        private readonly EventLevelCounter counter;
+        private readonly ILog log;
+
+        public LogLevelMetrics(EventLevelCounter counter, IMetricContext context, ILog log)
+        {
+            context.CreateMultiFuncGauge(ProvideMetrics, new FuncGaugeConfig {ScrapePeriod = ScrapePeriod});
```

Comment:
There's a race here. `CreateMultiFuncGauge` call effectively schedules the delegate (`ProvideMetrics`) for execution. Potentially it might be executed before the next statements and ctor and fail with NRE.